contracts/AuraLiquidityMigrator.sol

```diff
@@ -0,0 +1,194 @@
+        string memory symbol,
+        IERC20[] memory tokens,
+        uint256 swapFeePercentage,
+        address owner,
+        uint256 minOut
+    ) external returns (address pool) {
+        require(tokens.length == 2, "only token pairs");
+        // 1. Remove liquidity
+        (uint256 amount0, uint256 amount1) = _removeLiquidityUniswapV2(fromLpToken, user);
+
+        // 2. Create balancer pool
+        pool = _createBalancerPool(name, symbol, tokens, swapFeePercentage, owner);
+
+        // 3. Deposit to balancer pool
+        uint256[] memory maxAmountsIn = new uint256[](2);
+        maxAmountsIn[0] = amount0;
+        maxAmountsIn[1] = amount1;
+
+        _addLiquidityBalancer(pool, tokens, maxAmountsIn, user, minOut);
+
+        emit PoolCreated(pool);
+    }
+
+    /**
+     * @dev Migrates a liquidity position, adds liquidity to the pool, and stake the BPT
+     *
+     * @param fromLpToken The LP token to migrate
+     * @param user The owner of the lp position
+     * @param tokens The underlying tokens of the pool.
+     * @param minOut The min amount of bpt
+     * @param pool The pool address to add liquidity
+     * @param rewardPoolAddress The aura reward pool address
+     */
+    function migrateUniswapV2AndJoinPool(
+        address fromLpToken,
+        address user,
+        IERC20[] memory tokens,
+        uint256 minOut,
+        address pool,
+        address rewardPoolAddress
+    ) external {
+        require(tokens.length == 2, "only token pairs");
+
+        // 1. Remove liquidity
+        (uint256 amount0, uint256 amount1) = _removeLiquidityUniswapV2(fromLpToken, user);
+
+        // 2. Deposit to balancer pool
+        uint256[] memory maxAmountsIn = new uint256[](2);
+        maxAmountsIn[0] = amount0;
+        maxAmountsIn[1] = amount1;
+
+        _addLiquidityBalancer(pool, tokens, maxAmountsIn, address(this), minOut);
+
+        uint256 minted = IERC20(pool).balanceOf(address(this));
+        require(minted > 0, "!mint");
+
+        // 3. Deposit to reward pool
+        IERC20(pool).safeIncreaseAllowance(rewardPoolAddress, minted);
+        RewardPool(rewardPoolAddress).deposit(minted, user);
+    }
+
+    function _removeLiquidityUniswapV2(
+        address lpToken, //Uniswap LP token
+        address user
+    ) internal returns (uint256 amount0, uint256 amount1) {
+        // Remove liquidity position from IUniswapV2
+        IUniswapV2Pair pair = IUniswapV2Pair(lpToken);
+
+        uint256 liquidity = pair.balanceOf(user);
+        require(pair.transferFrom(msg.sender, address(pair), liquidity), "!liquidity");
+
+        (amount0, amount1) = pair.burn(address(this));
+    }
+
+    function _createBalancerPool(
+        string memory name,
+        string memory symbol,
+        IERC20[] memory tokens,
+        uint256 swapFeePercentage,
+        address owner
+    ) internal returns (address pool) {
+        uint256[] memory weights = new uint256[](2);
+        weights[0] = 500000000000000000; // 50%
```

Comment:
do these need to be scaled?

Reply:
Yes , I took as a reference the inputs on this tx that creates a balancer pool https://etherscan.io/tx/0x76ef3741298aec99c8cc88d5b773e601e8bc3ef75e6705eef6f1276b216084fa